docs/DeveloperGuide.md

```diff
@@ -250,26 +250,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
 Priority | As a ... | I want to ... | So that I can...
 -------- | :-------- | :--------- | :-----------
 `* * *` | new user | see usage instructions | refer to instructions when I forget how to use the App
-`* * *` | user | add a new person |
-`* * *` | user | delete a person | remove entries that I no longer need
-`* * *` | user | find a person by name | locate details of persons without having to go through the entire list
-`* *` | user | hide [private contact details](#private-contact-detail) by default | minimize chance of someone else seeing them by accident
-`*` | user with many persons in the address book | sort persons by name | locate a person easily
-
-{More to be added}
+`* * *` | user | add a new task using natural language | add a task to CMDo
+`* * *` | user | delete a task | remove entries that I no longer need
+`* * *` | user | edit a task| edit the task by accessing the task and typing the changes
+`* * *` | user | find a task by keywords | access details of tasks quickly without having to go through the entire list
+`* * *` | user |only view uncompleted task in CMDo| avoid being confused by completed task that are overdue| 
+`* * *` | user | my tasks sorted by due date and due time|locate urgent tasks easily | 
+`* * *` | user | remove my completed tasks| see only uncompleted tasks | 
+`* * *` | user | book by time slots| block out time for unconfirmed events |
+`* * *` | user | undo my previous action | make mistakes | 
+`* * *` | user | redo my previous action upon undo | make mistakes |
+`* * *` | user | set priority to my task | know which task is more important | 
+`* *` | user |Simply assign a date due to my todo by typing in the due date| to enter due dates easily
+`* *` | user |block out time slots of unconfirmed tasks| avoid scheduling tasks that clash
+`* *` | user |auto reschedule a task i am unable to complete due at the moment| save the effort of manual rescheduling
```

Comment:
this sounds cool! ~ is is a special feature u guys are deciding to have? if yes ~ lets talk more abt it during upcoming tutorial I do have some interesting ideas abt this ~ XD ~ however u have to rephrase it a little bcos it currently sounds a little like the "edit command" - you can do so by perhaps adding a simple "allow the system to auto....." <<< just a suggestion ~